examples/favorites_advanced/bloc_favorites_advanced/lib/feature_puppy/details/ui_components/puppy_details.dart

```diff
@@ -0,0 +1,91 @@
+import 'dart:io';
+
+import 'package:bloc_sample/feature_puppy/details/blocs/puppy_details_bloc.dart';
+import 'package:bloc_sample/feature_puppy/details/ui_components/puppy_details_app_bar.dart';
+import 'package:favorites_advanced_base/core.dart';
+import 'package:favorites_advanced_base/models.dart';
+import 'package:flutter/material.dart';
+import 'package:favorites_advanced_base/resources.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class PuppyDetails extends StatelessWidget {
+  const PuppyDetails({
+    required this.puppy,
+    Key? key,
+  }) : super(key: key);
+
+  final Puppy puppy;
+
+  @override
+  Widget build(BuildContext context) => SafeArea(
+        top: false,
+        bottom: false,
+        key: const ValueKey(Keys.puppyDetailsPage),
+        child: BlocProvider(
+          create: (context) => PuppyDetailsBloc(
+            coordinatorBloc: context.read(),
+            puppy: puppy,
+          ),
+          child: Stack(
+            children: [
+              BlocBuilder<PuppyDetailsBloc, PuppyDetailsState>(
+                builder: (context, state) => Hero(
+                  tag: '$PuppyCardAnimationTag ${puppy.id}',
+                  child: _buildPuppyBackgroundImage(
+                      state.imagePath ?? puppy.asset),
+                ),
+              ),
+              Padding(
+                padding: const EdgeInsets.all(16),
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  mainAxisAlignment: MainAxisAlignment.end,
+                  children: [
+                    const SizedBox(height: 10),
+                    BlocBuilder<PuppyDetailsBloc, PuppyDetailsState>(
+                        builder: (context, state) => Text(
+                              state.puppy?.name ?? puppy.name,
```

Comment:
Think of a way of avoiding all of those nullability checks

Reply:
I set a non-nullable `Puppy `in `PuppyDetailsState `and with that removed the nullability checks.